Source/items.cpp

```diff
@@ -1941,10 +1938,13 @@ void SpawnOnePremium(Item &premiumItem, int plvl, const Player &player)
 
 	plvl = clamp(plvl, 1, 30);
 
-	int count = 0;
+	int maxCount = 150;
+	if (!gbIsHellfire) {
+		// Diablo logic could lead to an infinite loop if a suitable item can never be generated, using an arbitrary cutoff here to let the game continue.
+		maxCount = 65000;

```

Comment:
This equates to roughly 200ms on my 3700x, that's probably plenty fine. But still can't help but be nervous about putting an arbitrary cutoff in there :sweat_smile:

Reply:
Could use max int if that makes you feel better 😄 No one is likely to let the game sit there for hours waiting for it to load town.

Reply:
I think 2000000 would make me calm, cause that would be enough that the game would timeout during multiplayer so there is no way that is an acceptable value :D

Reply:
Would it be better to keep the infinite loop?

```c++
const bool unlimited = !gbIsHellfire;
for (int count = 0; unlimited || count < maxCount; count++) {
	...
}
```

Reply:
If that change was applied then I think we can merge it and look in to applying a limit at a later point.